### ISSUE TITLE
version 3.0.0: sudden "team_is_restricted"

### ISSUE THREAD

**Issue by github-cygwin** [wee-slack/wee-slack#949]
Hi,

I'm running the wee-slack 3.0.0 version successfully for quite some time. Today, all of a sudden, during an active connection to an enterprise workspace, I'm seeing the word "disconnected" in the input line, right beside my nic name.  Funny enough I had written a message (via firefox, though) only an hour ago.

I restarted wee-chat, but wee-slack cannot connect anymore.  There's a new error message I never saw before:

> slack: Failed connecting to workspace "\<enterprise\>" Error from Slack API method client.counts with request None for workspace \<enterprise\>: {'ok': False, 'error': 'team_is_restricted'} (run `/slack debug error 6ec7a9c6-1f16-41d9-b53c-926a22e57aa8` for the stack trace)

The "team_is_restricted" is a new one.

The slack session in firefox was still active, but even so, I exited wee-chat again,
extracted a new token from firefox and changed it in slack.py accordingly.

Only the API cookie had changed, not the API token.

However, after restarting wee-chat, wee-slack still cannot connect, the above error persists.

Do you know what the problem here may be, and if I can do anything myself?

Thanks,
Corinna

**Comment by trygveaa** [comment]
Yeah, someone reported this on IRC too. I see that if you use the enterprise org token instead of the workspace token it still works.

If you use the `extract_token_from_browser.py` you can get this token by removing `not` on line 304, so it becomes `if team["id"].startswith("E")`, or if you're pulling it out of the browser yourself you can use `TS.boot_data.enterprise_api_token`.

I'll see about improving error handling for this. Maybe see if I can make it work with the workspace tokens, not sure. If enterprise org tokens work fine for everyone, might just require that.

Note that when you use an enterprise org token, you'll see all channels you're a member of across all the workspaces in the org.

**Comment by github-cygwin** [comment]
Yes, this works for me, too.

Great, thank you!
Corinna
